src/query/parser/promql/types.go

```diff
@@ -141,8 +141,10 @@ func NewFunctionExpr(name string, argValues []interface{}) (parser.Params, error
 		linear.MinuteType, linear.MonthType, linear.YearType:
 		return linear.NewDateOp(name)
 
-	case temporal.CountTemporalType:
-		return temporal.NewCountOp(argValues)
+	case temporal.AvgTemporalType, temporal.CountTemporalType, temporal.MinTemporalType,
```

Comment:
take it or leave it: All of these are getting pretty long, may be an idea to make it into a function that checks if it's one of these instead? Your call